### ISSUE TITLE
EventsRuleToLambda - does not add targets to Rules

### ISSUE THREAD

**Issue by knihit** [awslabs/aws-solutions-constructs#154]
`EventsRuleToLambda` works for schedule expression to trigger lambdas but if a custom event bus is passed with event patterns, it does not add the lambda function as the target to the rule.

```javascript
        const _target = new EventsRuleToLambda(this, 'RuleTargetLambda', {
            lambdaFunctionProps: props.target.lambdaFunctionProps,
            existingLambdaObj: props.target.existingLambda,
            eventRuleProps: {
                ...props.ingestionEventRuleProps,
                eventBus: this._bus,
                enabled: true
            }
        });
```

I had to add the following lines to make it work. My understanding it should be adding the lambda as the target within the construct.

```javascript
_target.eventsRule.addTarget(new targets.LambdaFunction(_target.lambdaFunction))
```
Is this expected behavior. 

### Environment

  - **CDK CLI Version      :** 1.93.0
  - **CDK Framework Version:** 1.93.0
  - **AWS Solutions Constructs Version     :** 1.93.0
  - **OS                   :** OSX
  - **Language             :** Typescript

### Other

--- 

This is :bug: Bug Report

**Comment by biffgaut** [comment]
Thanks, we'll take a look.

**Comment by jaresuth** [comment]
@biffgaut  - I believe this is related to specifying both `lambdaFunctionProps` and `existingLambdaObj` appears these should be used mutually exclusive from the comment [here](https://github.com/awslabs/aws-solutions-constructs/blob/main/source/patterns/%40aws-solutions-constructs/aws-events-rule-lambda/lib/index.ts#L25).  However, I am not seeing a scenario where it will throw an error if both of these specified and couldn't generate an error.  It is using a [aws-eventbridge-lambda](https://github.com/awslabs/aws-solutions-constructs/blob/0b018f2eb5a4acd8350867ec17a95dc2c6744545/source/patterns/%40aws-solutions-constructs/aws-eventbridge-lambda/lib/index.ts#L87-#L90)for taking `existingLambdaObj` props over  lambdaFunctionProps for `precedence`.  I don't think this is a bug and those should be used mutually exclusive.

**Comment by biffgaut** [comment]
Yes, since this was added to the backlog we've changed that behavior from "if both are passed the construct just ignores one" to "if both are passed it's an error" and centralized that code so that all constructs will observe that behavior. We've also explicitly added the ability to create/use custom event buses at the construct level rather than sneaking one in through the props. So the environment where this existed has totally changed. Before closing, I'd like to give a short time to hear from @knihit
